array_analyzer/extract/metadata.py

```diff
@@ -0,0 +1,221 @@
+import os
+import numpy as np
+from datetime import datetime
+import pandas as pd
+
+import array_analyzer.extract.txt_parser as txt_parser
+import array_analyzer.extract.constants as c
+
+
+class MetaData:
+
+    def __init__(self, input_folder_, output_folder_):
+        """
+        Parses metadata spreadsheets then populates all necessary ARRAY data structures
+        Extracts all necessary constants and assigns them in the constants.py namespace
+
+        :param input_folder_: str full path to metadata spreadsheet
+        :param output_folder_: str full path to output folder for reports and diagnostics
+        """
+
+        fiduc_, spots_, repl_, params_ = None, None, None, None
+
+        # parse fiducials, spot types, antigens, and hardware parameters from metadata
+        if c.METADATA_EXTENSION == 'xml':
+            # check that exactly one .xml exists
+            xml = [f for f in os.listdir(input_folder_) if '.xml' in f]
+            if len(xml) > 1:
+                raise IOError("more than one .xml file found, aborting")
+            xml_path = os.path.join(input_folder_, xml[0])
+
+            # parsing .xml
+            fiduc_, spots_, repl_, params_ = txt_parser.create_xml_dict(xml_path)
+
+        elif c.METADATA_EXTENSION == 'well':
+            self._set_run_path(output_folder_)
+            return
+
+        elif c.METADATA_EXTENSION == 'csv':
+            # check that three .csvs exist
+            three_csvs = ['array_format_antigen', 'array_format_type', 'array_parameters']
+            csvs = [f for f in os.listdir(input_folder_) if '.csv' in f]
```

Comment:
good question.  This is something i wrote before we settled on xlsx.  It is not enabled, and I haven't removed it just in case.  But probably it'll disappear soon.

Reply:
sorry, it is not used.  That requires a separate .csv for every sheet.  I thought it was going to be our approach, but I think excel .xlsx workbooks are better, especially with pandas